lib/fog/azurerm/models/network/subnet.rb

```diff
@@ -19,8 +19,16 @@ def self.parse(subnet)
           hash['resource_group'] = get_resource_group_from_id(subnet['id'])
           hash['virtual_network_name'] = get_virtual_network_from_id(subnet['id'])
           hash['address_prefix'] = subnet['properties']['addressPrefix']
-          hash['network_security_group_id'] = subnet['properties']['networkSecurityGroup']['id'] unless subnet['properties']['networkSecurityGroup'].nil?
-          hash['route_table_id'] = subnet['properties']['routeTable']['id'] unless subnet['properties']['routeTable'].nil?
+          hash['network_security_group_id'] = if subnet['properties']['networkSecurityGroup'].nil?
+                                                nil
+                                              else
+                                                subnet['properties']['networkSecurityGroup']['id']
+                                              end
+          hash['route_table_id'] = if subnet['properties']['routeTable'].nil?
+                                     nil
```

Comment:
Ternary operator here will increase the line length. Plus rubocop accepted this style.